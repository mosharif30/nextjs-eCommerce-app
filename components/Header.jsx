import React from 'react'
import Link from 'next/link'
/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react'

function Header() {
  const primary = '#2c3e50'
  return (
    <div>
      <navbar
        css={css`
          display: flex;
          background-color: ${primary};
          font-size: 24px;
          width: 100%;
          padding: 2%;
          overflow: hidden;
        `}
      >
        <span
          css={css`
            color: white;
          `}
        >
          فروشگاه اینترنتی نکست جی اس
        </span>
        <ul
          css={css`
            display: flex;
            list-style-type: none;
            color: white;
            margin: 0;
            width: 50%;
          `}
        >
          <li
            css={css`
              padding-right: 5%;
            `}
          >
            <Link href="/">خانه</Link>
          </li>
          <li
            css={css`
              padding-right: 5%;
            `}
          >
            <Link href="/about">درباره ما</Link>
          </li>
          <li
            css={css`
              padding-right: 5%;
            `}
          >
            <Link href="/contact">ارتباط با ما</Link>
          </li>
        </ul>
        <ul
          css={css`
            display: flex;
            list-style-type: none;
            color: white;
            margin-right: auto;
            margin-top: 0;
            margin-bottom: 0;
            width: 20%;
          `}
        >
          <li
            css={css`
              padding-right: 5%;
              margin-right: auto;
            `}
          >
            <Link href="/">سبد خرید</Link>
          </li>
          <li
            css={css`
              padding-right: 5%;
              margin-right: auto;
            `}
          >
            <Link href="/">ورود</Link>
          </li>
        </ul>
      </navbar>
    </div>
  )
}

export default Header
