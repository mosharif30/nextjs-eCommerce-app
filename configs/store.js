import { applyMiddleware, createStore } from 'redux'
import combineReducers from '../reducers'
import { composeWithDevTools } from 'redux-devtools-extension'
import thunk from 'redux-thunk'
const composeEnhancers = composeWithDevTools({
  name: 'book',
})
const store = createStore(
  combineReducers,
  composeEnhancers(applyMiddleware(thunk))
)

export default store
