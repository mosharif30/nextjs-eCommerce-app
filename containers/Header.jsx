import React from 'react'
import Link from 'next/link'
/** @jsxImportSource @emotion/react */
import { css, useTheme } from '@emotion/react'

function Header() {
  const theme = useTheme()

  return (
    <div>
      <navbar
        css={css`
          display: flex;
          background-color: ${theme.colors.primary};
          font-size: 24px;
          width: 100%;
          padding: 2%;
          overflow: hidden;
          @media (max-width: 1082px) {
            font-size: 20px;
          }
          @media (max-width: 706px) {
            flex-direction: column;
            align-items: center;
          }
        `}
      >
        <span
          css={css`
            color: white;
          `}
        >
          فروشگاه اینترنتی نکست
        </span>
        <ul
          css={css`
            display: flex;
            list-style-type: none;
            color: white;
            margin: 0;
            width: 50%;
            @media (max-width: 706px) {
              flex-direction: column;
              margin-top: 5%;
              align-items: center;
              padding-right: 0;
            }
          `}
        >
          <li
            css={css`
              padding-right: 5%;
            `}
          >
            <Link href="/">خانه</Link>
          </li>
          <li
            css={css`
              padding-right: 5%;
            `}
          >
            <Link href="/about">درباره ما</Link>
          </li>
          <li
            css={css`
              padding-right: 5%;
            `}
          >
            <Link href="/contact">ارتباط با ما</Link>
          </li>
        </ul>
        <ul
          css={css`
            display: flex;
            list-style-type: none;
            color: white;
            margin-right: auto;
            margin-top: 0;
            margin-bottom: 0;
            width: 20%;
            @media (max-width: 1082px) {
              font-size: 20px;
              padding: 2px;
            }
            @media (max-width: 706px) {
              flex-direction: row;
              justify-content: space-between;
              width: 100%;
            }
          `}
        >
          <li
            css={css`
              padding-right: 5%;
              margin-right: auto;
              @media (max-width: 706px) {
                margin-right: 0;
                padding-right: 0;
              }
            `}
          >
            <Link href="/">سبد خرید</Link>
          </li>
          <li
            css={css`
              padding-right: 5%;
              margin-right: auto;
            `}
          >
            <Link href="/login">ورود</Link>
          </li>
        </ul>
      </navbar>
    </div>
  )
}

export default Header
