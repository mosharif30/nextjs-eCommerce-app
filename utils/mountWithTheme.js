import { ThemeProvider } from '@emotion/react'
import theme from '../configs/theme'
const mountWithTheme 
