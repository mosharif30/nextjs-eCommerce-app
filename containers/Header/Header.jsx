import React, { useEffect } from 'react'
import Link from 'next/link'
/** @jsxImportSource @emotion/react */
import { css, useTheme } from '@emotion/react'
import { useDispatch, useSelector } from 'react-redux'
import { Button } from '../../components'
import { getTotals } from '../../reducers/cartSlice'

function Header() {
  const dispatch = useDispatch()
  const cart = useSelector((state) => state.cart)

  useEffect(() => {
    dispatch(getTotals())
  }, [cart, dispatch])
  const auth = useSelector((state) => state.auth)

  const theme = useTheme()
  const { cartTotalQuantity } = useSelector((state) => state.cart)
  return (
    <div>
      <div
        css={css`
          display: flex;
          padding: 1%;
          background-color: ${theme.colors.primary};

          @media (max-width: 706px) {
            flex-direction: column;
            align-items: center;
          }
        `}
      >
        <div
          css={css`
            display: flex;

            color: ${theme.colors.text};
            width: 25%;
            align-items: center;
            justify-content: center;
            @media (max-width: 706px) {
              width: 100%;
              justify-content: center;
            }
          `}
        >
          <img
            css={css`
              width: 120px;
            `}
            src="/nextl.png"
            alt="logo"
          />{' '}
        </div>
        <ul
          css={css`
            display: flex;
            list-style-type: none;
            color: ${theme.colors.text};
            width: 40%;
            @media (max-width: 706px) {
              flex-direction: column;
              margin-top: 5%;
              align-items: center;
              padding-right: 0;
            }
          `}
        >
          <li
            css={css`
              display: flex;

              align-items: center;
              justify-content: center;
            `}
          >
            <Link href="/">خانه</Link>
          </li>
          <li
            css={css`
              display: flex;

              padding-right: 5%;
              align-items: center;
              justify-content: center;
              @media (max-width: 706px) {
                padding-right: 0;
              }
            `}
          >
            <Link href="/about">درباره ما</Link>
          </li>
          <li
            css={css`
              display: flex;

              padding-right: 5%;
              align-items: center;
              justify-content: center;
              @media (max-width: 706px) {
                padding-right: 0;
              }
            `}
          >
            <Link href="/contact">ارتباط با ما</Link>
          </li>
        </ul>
        <ul
          css={css`
            display: flex;
            justify-content: space-between;
            list-style-type: none;
            color: ${theme.colors.text};
            margin-right: auto;
            width: 40%;
            @media (max-width: 1082px) {
              padding: 2px;
            }
            @media (max-width: 706px) {
              flex-direction: row;
              justify-content: space-between;
              width: 100%;
            }
          `}
        >
          <li
            css={css`
              display: flex;

              padding-right: 5%;
              align-items: center;
              margin-right: auto;
              margin-left: 5%;
              @media (max-width: 706px) {
                margin-right: 0;
                padding-right: 0;
              }
            `}
          >
            <Link href="/cart">
              <Button nohover>
                <div
                  css={css`
                    display: flex;
                  `}
                >
                  <span>
                    <svg
                      xmlns="http://www.w3.org/2000/svg"
                      width="30"
                      height="30"
                      viewBox="0 0 24 24"
                    >
                      <path d="M10 19.5c0 .829-.672 1.5-1.5 1.5s-1.5-.671-1.5-1.5c0-.828.672-1.5 1.5-1.5s1.5.672 1.5 1.5zm3.5-1.5c-.828 0-1.5.671-1.5 1.5s.672 1.5 1.5 1.5 1.5-.671 1.5-1.5c0-.828-.672-1.5-1.5-1.5zm1.336-5l1.977-7h-16.813l2.938 7h11.898zm4.969-10l-3.432 12h-12.597l.839 2h13.239l3.474-12h1.929l.743-2h-4.195z" />
                    </svg>
                  </span>
                  <span
                    css={css`
                   background-color:${theme.colors.secondary};
                   padding:20%;
                   color:white;
                  border-radius: 100%;
                    }
                  `}
                  >
                    {cartTotalQuantity}
                  </span>
                </div>
              </Button>
            </Link>
          </li>
          {(!auth || !auth.logged) && (
            <>
              <li
                css={css`
                  display: flex;

                  padding-right: 5%;
                  align-items: center;
                  margin-right: auto;
                  margin-left: 5%;
                `}
              >
                <Link href="/login">ورود</Link>
              </li>
              <li
                css={css`
                  display: flex;

                  padding-right: 5%;
                  align-items: center;
                  padding: 5%;
                  border-radius: 5%;
                  background-color: ${theme.colors.dark};
                  color: white;
                `}
              >
                <Link href="/register">ثبت نام</Link>
              </li>
            </>
          )}
          {auth && auth.logged && (
            <>
              <li
                css={css`
                  display: flex;

                  padding-right: 5%;
                  align-items: center;
                  padding: 5%;
                  border-radius: 5%;
                  background-color: ${theme.colors.secondary};
                  color: white;
                `}
              >
                <div> خوش آمدید </div>
              </li>
            </>
          )}
        </ul>
      </div>
    </div>
  )
}

export default Header
