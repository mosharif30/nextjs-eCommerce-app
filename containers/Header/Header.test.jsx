import mountWithTheme from '../../utils/mountWithTheme'
import Header from './Header'
import { Provider } from 'react-redux'
import store from '../../configs/store'
describe('Header', () => {
  test('render correctly', () => {
    const component = mountWithTheme(
      <Provider store={store}>
        <Header></Header>
      </Provider>
    )
    const tree = component
    expect(tree).toMatchSnapshot()
  })
})
