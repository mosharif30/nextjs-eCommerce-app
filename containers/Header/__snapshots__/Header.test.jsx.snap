// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render correctly 1`] = `
<div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-14j6nrd-Header{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding:1%;background-color:#ecf0f1;}@media (max-width: 706px){.css-14j6nrd-Header{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}}",
      }
    }
    data-emotion="css 14j6nrd-Header"
  />
  <div
    className="css-14j6nrd-Header"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-1rljxef-Header{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;color:#2c3e50;width:25%;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;}@media (max-width: 706px){.css-1rljxef-Header{width:100%;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;}}",
        }
      }
      data-emotion="css 1rljxef-Header"
    />
    <div
      className="css-1rljxef-Header"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-d59lwj-Header{width:120px;}",
          }
        }
        data-emotion="css d59lwj-Header"
      />
      <img
        alt="logo"
        className="css-d59lwj-Header"
        src="/nextl.png"
      />
       
    </div>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-qae97p-Header{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style-type:none;color:#2c3e50;width:40%;}@media (max-width: 706px){.css-qae97p-Header{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;margin-top:5%;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:0;}}",
        }
      }
      data-emotion="css qae97p-Header"
    />
    <ul
      className="css-qae97p-Header"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-j5tgcz-Header{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;}",
          }
        }
        data-emotion="css j5tgcz-Header"
      />
      <li
        className="css-j5tgcz-Header"
      >
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          خانه
        </a>
      </li>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-mg5t0m-Header{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-right:5%;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;}@media (max-width: 706px){.css-mg5t0m-Header{padding-right:0;}}",
          }
        }
        data-emotion="css mg5t0m-Header"
      />
      <li
        className="css-mg5t0m-Header"
      >
        <a
          href="/about"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          درباره ما
        </a>
      </li>
      <li
        className="css-mg5t0m-Header"
      >
        <a
          href="/contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          ارتباط با ما
        </a>
      </li>
    </ul>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-13uuzsx-Header{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;justify-content:space-between;list-style-type:none;color:#2c3e50;margin-right:auto;width:40%;}@media (max-width: 1082px){.css-13uuzsx-Header{padding:2px;}}@media (max-width: 706px){.css-13uuzsx-Header{-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;justify-content:space-between;width:100%;}}",
        }
      }
      data-emotion="css 13uuzsx-Header"
    />
    <ul
      className="css-13uuzsx-Header"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1vencho-Header{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-right:5%;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;margin-right:auto;margin-left:5%;}@media (max-width: 706px){.css-1vencho-Header{margin-right:0;padding-right:0;}}",
          }
        }
        data-emotion="css 1vencho-Header"
      />
      <li
        className="css-1vencho-Header"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-24zwvv-Button{width:100%;background-color:#ecf0f1;color:#2c3e50;font-size:1em;font-family:'Vazir';-webkit-text-decoration:none;text-decoration:none;padding:16px;border:2px;display:inline-block;position:relative;cursor:pointer;transition-duration:0.4s;}.css-24zwvv-Button:hover{box-shadow:#ecf0f1;background-color:#ecf0f1;}",
            }
          }
          data-emotion="css 24zwvv-Button"
        />
        <button
          className="css-24zwvv-Button"
          onClick={[Function]}
          type="button"
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-x6egge-Header{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
              }
            }
            data-emotion="css x6egge-Header"
          />
          <div
            className="css-x6egge-Header"
          >
            <span>
              <svg
                height="30"
                viewBox="0 0 24 24"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 19.5c0 .829-.672 1.5-1.5 1.5s-1.5-.671-1.5-1.5c0-.828.672-1.5 1.5-1.5s1.5.672 1.5 1.5zm3.5-1.5c-.828 0-1.5.671-1.5 1.5s.672 1.5 1.5 1.5 1.5-.671 1.5-1.5c0-.828-.672-1.5-1.5-1.5zm1.336-5l1.977-7h-16.813l2.938 7h11.898zm4.969-10l-3.432 12h-12.597l.839 2h13.239l3.474-12h1.929l.743-2h-4.195z"
                />
              </svg>
            </span>
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".css-65npxb-Header{background-color:#16a085;padding:20%;color:white;border-radius:100%;}",
                }
              }
              data-emotion="css 65npxb-Header"
            />
            <span
              className="css-65npxb-Header"
            >
              0
            </span>
          </div>
        </button>
      </li>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-ur5efn-Header{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-right:5%;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;margin-right:auto;margin-left:5%;}",
          }
        }
        data-emotion="css ur5efn-Header"
      />
      <li
        className="css-ur5efn-Header"
      >
        <a
          href="/login"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          ورود
        </a>
      </li>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-dv6tsd-Header{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-right:5%;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding:5%;border-radius:5%;background-color:#34495e;color:white;}",
          }
        }
        data-emotion="css dv6tsd-Header"
      />
      <li
        className="css-dv6tsd-Header"
      >
        <a
          href="/register"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          ثبت نام
        </a>
      </li>
    </ul>
  </div>
</div>
`;
